src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -120,38 +120,45 @@ doTxHelp(RPC::Context& context, TxArgs const& args)
     }
 
     std::shared_ptr<Transaction> txn;
+    std::shared_ptr<TxMeta> meta;
     auto ec{rpcSUCCESS};
 
+    using TxPair =
+        std::pair<std::shared_ptr<Transaction>, std::shared_ptr<TxMeta>>;
+
     result.searchedAll = SearchedAll::unknown;
     if (args.ledgerRange)
     {
-        boost::variant<std::shared_ptr<Transaction>, bool> v =
-            context.app.getMasterTransaction().fetch(args.hash, range, ec);
+        auto v = context.app.getMasterTransaction().fetch(args.hash, range, ec);
+
+        if (!v)
+            return {result, rpcTXN_NOT_FOUND};
 
-        if (v.which() == 1)
+        if (auto b = std::get_if<bool>(&v.value()))
         {
-            result.searchedAll =
-                boost::get<bool>(v) ? SearchedAll::yes : SearchedAll::no;
+            result.searchedAll = *b ? SearchedAll::yes : SearchedAll::no;
+
             return {result, rpcTXN_NOT_FOUND};
         }
         else
         {
-            txn = boost::get<std::shared_ptr<Transaction>>(v);
+            std::tie(txn, meta) = std::get<TxPair>(v.value());
         }
     }
     else
     {
-        txn = context.app.getMasterTransaction().fetch(args.hash, ec);
+        auto v = context.app.getMasterTransaction().fetch(args.hash, ec);
+
+        if (!v)
+            return {result, rpcTXN_NOT_FOUND};
+
+        std::tie(txn, meta) = v.value();
     }
 
     if (ec == rpcDB_DESERIALIZATION)
     {
         return {result, ec};
     }
-    if (!txn)
-    {
-        return {result, rpcTXN_NOT_FOUND};
-    }
```

Comment:
I wouldn't lose the checks for null txn and null meta. Right now the code is correct, but given the return type of `fetch` is likely to be refactored, and we don't know what that refactoring will be, I'd prefer to leave the checks for null.